analyzer/src/funTest/kotlin/BabelTest.kt

```diff
@@ -32,6 +32,7 @@ import com.here.ort.utils.test.patchExpectedResult
 
 import io.kotlintest.TestCase
 import io.kotlintest.TestResult
+import io.kotlintest.matchers.maps.shouldContainKey
```

Comment:
I don't like how this change makes the test code longer and harder to read, because of the many `let`s. Is it an option to put it in another PR and discuss it separately?

Commit message:
* s/diffcult/difficult
* "makes it diffcult to ensure Identifier.toString() is not called for serialization" - Is this really about serialization? I expected it to be about making sure `toCoordinates` is now called when identifiers are used in strings.
* You are doing more changes than avoiding `shouldNotBe null`, e.g. you also avoid `beEmpty()` and `containAll()` and align variable names.

Reply:
As discussed offline, I'll simply drop the last commit as it was anyway only needed to prove that `toString()` does not get called anymore.